File: ruby/series/series.rb
class Series
  attr_reader :digits

  def initialize(number)
    @digits = number.chars.map(&:to_i)
  end

  def slices(slice_size)
    raise ArgumentError, "It's too large" if digits.size < slice_size
    digits.each_cons(slice_size).to_a
  end
end

File: python/series/series.py
def slices(string, slice_size):
    numbers = [int(char) for char in list(string)]
    last_idx = len(numbers) + 1 - slice_size
    if last_idx < 1 or len(numbers) < last_idx:
        raise ValueError

    return [numbers[i:i+slice_size] for i in range(0, last_idx)]
